### DISCUSSION TITLE
how can I add a text box in c# and bind the font size to my adaptive trigger resource

### DISCUSSION THREAD

**Discussion by ashah7** [unoplatform/uno.toolkit.ui#1307] [General]
I've added a textbox and made the font size responsive by adding the "FontSize" binding like the below.
Using adaptive triggers this works nicely for different screen sizes in xaml

Is there a way I can add the `FontSize ` binding of my textbox using c# only ?

thanks

`<TextBlock x:Name="CustomMyFont" Text="This is a sample" FontSize="{Binding FontSize, ElementName=CustomMyFont, Mode=TwoWay}"/>`

```

 <VisualState x:Name="TenEightyP">
   <VisualState.StateTriggers>
     <AdaptiveTrigger MinWindowWidth="1920"/>
   </VisualState.StateTriggers>
   <VisualState.Setters>
     <Setter Target="CustomMyFont.FontSize" Value="35"/>
   </VisualState.Setters>
 </VisualState>
```

**Comment by Xiaoy312** [comment]
xaml:
```xml
<TextBlock x:Name="CodeBehindExample" Text="This is another way of doing it" />
```

code-behind:
```cs
	private int _lastWidthBreakPoint = -1;

	public MainPage()
	{
		this.InitializeComponent();
		this.SizeChanged += (s, e) =>
		{
			var breakpoint = e.NewSize.Width switch
			{
				var x when x >= 1920 => 1920,
				//var x when x >= 1080 => 1080, // if you have more breakpoints, add like so
				_ => 0,
			};
			if (breakpoint != _lastWidthBreakPoint)
			{
				if (breakpoint == 0)
				{
					CodeBehindExample.FontSize = 10; // note: you may want to match this value with default textbox fontsize
				}
				//else if (breakpoint == 1080) { } // if you have more breakpoints, add like so
				else if (breakpoint == 1920)
				{
					CodeBehindExample.FontSize = 35;
				}

				// store the last used breakpoint, so we only update when moving to another breakpoint
				_lastWidthBreakPoint = breakpoint;
			}
		};
	}
```

**Comment by Xiaoy312** [comment]
Additionally, you can also use [the responsive extension from toolkit](https://platform.uno/docs/articles/external/uno.toolkit.ui/doc/helpers/responsive-extension.html):
```xml
xmlns:utu="using:Uno.Toolkit.UI"
...

<TextBlock Text="This is another way of doing it" FontSize="{utu:Responsive Normal=10, Widest=35}">
	<TextBlock.Resources>
		<utu:ResponsiveLayout x:Key="WidestAt1920" Widest="1920" />
	</TextBlock.Resources>
</TextBlock>
```

note1: Here, I am using a local layout override at the control level for simplicity, but you can also define this globally.
note2: You want to make sure toolkit is included if not already:
- for newer project, just add toolkit to the `UnoFeatures` property:

		<UnoFeatures>
		  	toolkit;
		</UnoFeatures>

- for older project, just a package reference to [`Uno.Toolkit.WinUI`](https://www.nuget.org/packages/uno.toolkit.winui) or [`Uno.Toolkit.UI`](https://www.nuget.org/packages/uno.toolkit.ui) for uwp

note3: this feature is not supported on uwp (for windows)

**Comment by ashah7** [comment]
thank you this helped to solve my problem
